### PR TITLE
[#104] update dockerfile

### PR DESCRIPTION
**Pull request by rvanbutselaar**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17.2-alpine3.14 as builder
+FROM golang:1.19.0 as builder
```
Comment (030) [inline Dockerfile:1]: Could you change this to: `1.19.0-alpine3.16`?

File: cmd/dip/image.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ to quickly create a Cobra application.`,
 			if err != nil {
 				log.Fatal(err)
 			}
-			fmt.Println(latestTag)
```
Comment (030) [inline cmd/dip/image.go:32]: fmt is used to ensure that only the tag is returned, e.g.:

```
kdiutd.dip image --name sonatype/nexus3 --regex 3.*
3.41.0
```

Perhaps you could add a comment to ensure that the fmt is kept.

File: cmd/dip/image.go [hunk 3]
```diff
@@ -55,7 +55,13 @@ to quickly create a Cobra application.`,
 							log.Fatal(err)
 						}
 					}
+					log.Fatal(err)
+				}
+			}
 
+			if updateDockerfile {
+				err := docker.UpdateFROMStatementDockerfile(name, latestTag)
```
Comment (030) [inline cmd/dip/image.go:63]: The if and err could be combined if there is only one return, i.e.:

```
if err := docker.UpdateFROMStatementDockerfile(name, latestTag); err!=nil {
  log.Fatal(err)
}
```

File: internal/docker/docker.go [hunk 4]
```diff
@@ -34,3 +34,22 @@ func FileLatest(image, latestTag string) error {
...
+	if err != nil {
+		return err
+	}
+
+	re, err := regexp.Compile(`FROM ` + image + `:([a-z0-9\.-]+)`)
+	if err != nil {
+		return err
+	}
+	replaced := re.ReplaceAll(b, []byte(`FROM `+image+`:`+latestTag+`${2}`))
+	if err = os.WriteFile("Dockerfile", replaced, 0644); err != nil {
```
Comment (030) [inline internal/docker/docker.go:50]: `if err :=`

**Review by 030 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (rvanbutselaar) [reply Dockerfile]: Changed the regular expression in the docs to fix this

**Comment by rvanbutselaar** [comment]
Closes #104

**Pull request merged**